pkg/runtime/runtime.go

```diff
@@ -283,6 +293,39 @@ func (a *DaprRuntime) initRuntime(opts *runtimeOpts) error {
 		return err
 	}
 	a.namespace = a.getNamespace()
+
+	if a.runtimeConfig.ApplicationPort != 0 {
+		a.podInfo.ApplicationProbingPort = a.runtimeConfig.ApplicationPort
+	}
+
+	if a.runtimeConfig.Mode == modes.KubernetesMode {
```

Comment:
This entire `if` block can go inside the sidecar injector. This will not require the sidecar to have RBAC access to GET pods, which is uncalled for. The sidecar injector can inspect pod on the incoming injection request, and thus also save a call to the Kubernetes API. The information that the sidecar needs, `podInfo`, can be set by the injector on the Dapr sidecar as environment variables.

Reply:
But it still needs the RBAC access to GET pods when fetching the status of the containers?

Reply:
And I made a test on the injector that it cannot get the pod name when injecting the sidecar.

Reply:
I would really rather think of a design where the sidecar doesn't need to make GET requests for pod information. This will not play well with enterprise security requirements.

Reply:
I can scan the ports of the app container to determine the liveness of it, but I wonder whether it is enough because one might not have a port exposed?

Reply:
@yaron2 I have removed the logic that checks container status and the role binding.